File: src/main/java/it/ipzs/pidprovider/dto/TokenResponse.java
package it.ipzs.pidprovider.dto;

import java.io.Serializable;

import com.fasterxml.jackson.annotation.JsonProperty;

import lombok.Builder;
import lombok.Data;

@Data
@Builder
public class TokenResponse implements Serializable {

	private static final long serialVersionUID = -7979455405188597114L;

	@JsonProperty("access_token")
	private String accessToken;

	@JsonProperty("token_type")
	private String tokenType;

	@JsonProperty("expires_in")
	private int expiresIn;

	@JsonProperty("c_nonce")
	private String nonce;

	@JsonProperty("c_nonce_expires_in")
	private int nonceExpiresIn;

}

File: sdk-android/androidpidprovider/src/main/java/it/ipzs/androidpidprovider/network/response/TokenResponse.kt
@file:Suppress("unused")

package it.ipzs.androidpidprovider.network.response


import com.google.gson.annotations.SerializedName
import com.google.gson.annotations.Expose
import java.io.Serializable

internal class TokenResponse: Serializable{

    @SerializedName("access_token")
    @Expose
    val accessToken: String? = null

    @SerializedName("token_type")
    @Expose
    val tokenType: String? = null

    @SerializedName("expires_in")
    @Expose
    val expiresIn: Int? = null

    @SerializedName("c_nonce")
    @Expose
    val cNonce: String? = null

    @SerializedName("c_nonce_expires_in")
    @Expose
    val cNonceExpiresIn: Int? = null

}